### ISSUE TITLE
bug(datetime) - Datetime now returns an object, instead of an ISO string

### ISSUE THREAD

**Issue by zakton5** [driftyco/ionic#11175]
**Ionic version:**  (check one with "x")
[ ] **1.x**
[ ] **2.x**
[x] **3.x**

**I'm submitting a ...**  (check one with "x")
[x] bug report
[ ] feature request
[ ] support request

**Current behavior:**
Selecting a date with the datetime and clicking 'OK' returns an object to the model instead of an ISO string. According to the docs I should be receiving an ISO string: [DateTime Docs](http://ionicframework.com/docs/api/components/datetime/DateTime/)

**Expected behavior:**
I expect the datetime to return an ISO string as it did in ionic 2

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):
Cordova CLI: 6.5.0 
Ionic Framework Version: 3.0.1-201704062237
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.2.0
Ionic App Scripts Version: 1.3.0
ios-deploy version: 1.9.1 
ios-sim version: 5.0.13 
OS: macOS Sierra
Node Version: v6.10.1
Xcode version: Xcode 8.3.1 Build version 8E1000a

**Comment by jgw96** [comment]
Hello, thanks for using Ionic! Could you provide a repo or plunker we could use to reproduce this issue?

**Comment by sachithd** [comment]
This is exactly what I reported on
https://github.com/driftyco/ionic/commit/99142f8

on version npm install --save --save-exact ionic-angular@3.0.1-201704062237

**Comment by zakton5** [comment]
@jgw96 
Load up the ionic conference app. Download the current nightly. This bug is NOT on 3.0.1, only the build that @sachithd provided.
On line 28 in `about.html`, insert this: `{{conferenceDate}}`
Serve the app, go to About, change the date, and observe that it says `[object Object]`

**Comment by zakton5** [comment]
Also, this issue is labeled as v2 when it should be v3

**Comment by jgw96** [comment]
Hello, @zakton5 is this still an issue with [ionic-angular 3.1.0](https://github.com/driftyco/ionic/blob/master/CHANGELOG.md#310-2017-04-26) ?

**Comment by growthclick** [comment]
@jgw96 This is an error impacting 3.1.0 - I'm finding my datetime models are getting changed from string format (e.g. "08:00") to an object as in {ampm:"pm", hour:13, minute:0}. 

Are there any quick fixes to get back to date strings?

**Comment by growthclick** [comment]
Here is some more background on how I'm using ion-datetime with Ionic 3.1:  

 
<ion-datetime displayFormat="h:mm a"   pickerFormat="h:mm a" [(ngModel)]="user.preferences.dndAutoStartTime"  (ionChange)="saveUserDNDSettings()"></ion-datetime>
 

Ionic info system information:

Cordova CLI: 6.3.1 
Ionic Framework Version: 3.1.0
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.2.0
Ionic App Scripts Version: 1.3.5
ios-deploy version: 1.9.1 
ios-sim version: 5.0.3 
OS: macOS Sierra
Node Version: v6.9.1
Xcode version: Xcode 8.3.2 Build version 8E2002

**Comment by meeximum** [comment]
I have the same problem? ion-datetime returning an object instead a string :-/

**Comment by meeximum** [comment]
Maybe it's a problem within the onChange method?
The data seems to be a Ionics DateTimeData object, which normally should be serialized as ISO string?
[https://github.com/driftyco/ionic/blob/24544e01e150bf59bfc8a6d1cc38b7cdae860c4f/src/components/datetime/datetime.ts#L850]

**Comment by meeximum** [comment]
As workaround you can react on the ionChange event
`(ionChange)="onChangeDate($event)"`

`onChangeDate(dateTime: DateTime) {
    console.log(dateTime._text);
  }`

**Comment by growthclick** [comment]
Thanks for the workaround. That might work well with single binding on  [ngModel]. I retested with the component inside a form and with correct names but still had the same results.  

It looks like the object may be getting passed back in the onChange function without the convertDataToISO() call that is on line 836.  

https://github.com/driftyco/ionic/blob/24544e01e150bf59bfc8a6d1cc38b7cdae860c4f/src/components/datetime/datetime.ts#L850

https://github.com/driftyco/ionic/blob/24544e01e150bf59bfc8a6d1cc38b7cdae860c4f/src/components/datetime/datetime.ts#L836

**Comment by leoruhland** [comment]
```

**Comment by leoruhland** [comment]
I agree with @growthclick

I use datetime with formControlName and this saved me:
````
import { convertDataToISO } from 'ionic-angular/util/datetime-util';

const formatedDate = convertDataToISO(this.form.get('date_field').value);
````

**Comment by leoruhland** [comment]
Replying to @jgw96:
> is this still an issue with ionic-angular 3.1.0 ?

Yes, 3.0.1 release is ok and 3.1.0 release is missing convertDataToISO in DateTime component and has the bug.

**Comment by sachithd** [comment]
Any update on this please? I first reported this bug on 7th of April. Thanks

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

On Wed, Apr 26, 2017 at 11:21 PM, Leonardo Ruhland <notifications@github.com

**Comment by leoruhland** [comment]
@manucorporat fixed this on https://github.com/driftyco/ionic/commit/6677d80343b04c727cd30391f2b010aa3f056c22!
Just update to 3.1.1 for a fixed version.

Datetime always return a string right now. (https://github.com/driftyco/ionic/commit/aff2f08f9aa48319def50d4a9baea86e8cb7374a)

**Comment by topalavlad** [comment]
This same issue seems to be present in ionic 2.2.3

**Comment by jgw96** [comment]
Hello all! As this issue has been fixed in Ionic 3.1.1 I am going to close this issue for now. Thanks everyone!
